client/src/templates/Challenges/redux/current-challenge-saga.js

```diff
@@ -16,14 +18,20 @@ import { post } from '../../../utils/ajax';
 import { randomCompliment } from '../utils/get-words';
 import { updateSuccessMessage } from './';
 
-function* currentChallengeSaga({ payload }) {
+export const CURRENT_CHALLENGE_KEY = 'currentChallengeId';
+
+export function* currentChallengeSaga({ payload: id }) {
   const isSignedIn = yield select(isSignedInSelector);
   const currentChallengeId = yield select(currentChallengeIdSelector);
-  if (isSignedIn && payload !== currentChallengeId) {
+  // TODO: should the server update take place on challenge completion instead?
+  if (isSignedIn && id !== currentChallengeId) {
+    store.set(CURRENT_CHALLENGE_KEY, id);
+    yield put(updateCurrentChallengeId(id));
```

Comment:
`challengeMounted` doesn't have access to the same `state` as `fetchUserComplete`, does it?  Is there a good way to communicate between the independent `state`s other than by firing off a second action?

Reply:
All reducers receive all actions. Add `challengeMounted` action processing in the main reducer.

Reply:
Okay, so doing it this way means that 
```js
id !== currentChallengeId
```
is always false, because `challengeMounted` will have already updated `currentChallengeId` by this point.  Hence the server never gets updated.

I see two options.
1) Remove `id !== currentChallengeId` and accept that the server is going to get more unnecessary requests.
2) Use `updateCurrentChallengeId` as before.

Unless there's a third option that I'm missing, which do you think is best?

Reply:
Currently, we can safely remove this check, it did not work properly, because the `currentChallengeId` was not updated during the session.

Reply:
I was worried it would ping the server each time you reloaded a challenge.  However, that should not happen because the challenge gets mounted before the user comes back from the server.  i.e. `isSignedIn` is false, so the second check doesn't matter.